### PR TITLE
BUG: is_bool_dtype raises AttributeError when checking categorical Series

### LINKED ISSUE #45615: BUG: is_bool_dtype 'Series' object has no attribute 'categories'
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [x] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
from pandas import Series
from pandas.core.dtypes.common import is_bool_dtype

s = Series(['foo', 'bar', 'baz'], dtype='category')
is_bool_dtype(s)
```

### Issue Description

In pandas v1.3.0 the `is_bool_dtype` function returns `False` when passed a categorical `Series`
In pandas v1.4.0 it raises the error :
```
Traceback (most recent call last):
  File "./pandas/core/dtypes/common.py", line 1322, in is_bool_dtype
    arr_or_dtype = arr_or_dtype.categories
  File "./pandas/core/generic.py", line 5583, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'categories'
```

### Expected Behavior

Expect `is_bool_dtype` to return `False` for categorical Series as happened in previous pandas versions

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : bb1f651536508cdfef8550f93ace7849b00046ee
python           : 3.9.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-96-generic
Version          : #109-Ubuntu SMP Wed Jan 12 16:49:16 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8
pandas           : 1.4.0
numpy            : 1.21.3
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 20.2.4
setuptools       : 59.6.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.3.7
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : 5.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
</details>

### PR DESCRIPTION
**Pull request by Robbie-Palmer**
- [X] closes #45615
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.4.1.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ Fixed regressions
 Bug fixes
 ~~~~~~~~~
 - Fixed segfault in :meth:``DataFrame.to_json`` when dumping tz-aware datetimes in Python 3.10 (:issue:`42130`)
--
+- Bug in :meth:`common.is_bool_dtype` was raising an AttributeError when evaluating a categorical Series (:issue:`45615`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.4.1.rst:27]: can you reference this as api.types.is_bool_dtype

File: pandas/tests/dtypes/test_common.py [hunk 2]
```diff
@@ -562,6 +562,7 @@ def test_is_bool_dtype():
     assert not com.is_bool_dtype(int)
     assert not com.is_bool_dtype(str)
     assert not com.is_bool_dtype(pd.Series([1, 2]))
+    assert not com.is_bool_dtype(pd.Series(["a", "b"], dtype="category"))
```
Comment (jbrockmendel) [inline pandas/tests/dtypes/test_common.py:565]: do we have a categorical case where it is True?

Comment (phofl) [reply pandas/tests/dtypes/test_common.py:565]: We should add something like

``pd.Series([True, False], dtype='category')``

**Review by phofl — APPROVED** [review]

**Review by jbrockmendel — APPROVED** [review]

**Comment by phofl** [comment]
Could you merge master?

File: doc/source/whatsnew/v1.4.1.rst [hunk 3]
```diff
@@ -24,7 +24,7 @@ Fixed regressions
 Bug fixes
 ~~~~~~~~~
 - Fixed segfault in :meth:``DataFrame.to_json`` when dumping tz-aware datetimes in Python 3.10 (:issue:`42130`)
--
+- Bug in :meth:`api.types.is_bool_dtype` was raising an AttributeError when evaluating a categorical Series (:issue:`45615`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.1.rst:27]: normally we would use backticks around AttributeError and Series (as it highlites them). if you can update and ping on green.

**Comment by jreback** [comment]
also pls merge master

File: doc/source/whatsnew/v1.4.1.rst (hunk 3 above)
Comment (Robbie-Palmer) [reply doc/source/whatsnew/v1.4.1.rst]: Any advice on how to handle the failing checks?

**Comment by jreback** [comment]
thanks @Robbie-Palmer

**Comment by jreback** [comment]
@meeseeksdev backport 1.4.x

**Pull request merged**